### ISSUE TITLE
[BUG] Setting Parameter `rag_pipeline_feature_enabled: true` is not effect in 2.12.0 when restarting the cluster

### ISSUE THREAD

**Issue by lihuimingxs** [opensearch-project/ml-commons#2169]
**What is the bug?**

In Opensearch 2.12.0, I have set the parameter `plugins.ml_commons.rag_pipeline_feature_enabled` value to `true`, like is:

```
PUT _cluster/settings
{
  "persistent": {
    "plugins.ml_commons.rag_pipeline_feature_enabled": true
  }
}
```

Then I created a RAG pipline:

```
PUT /_search/pipeline/rag_demo
{
  "response_processors": [
    {
      "retrieval_augmented_generation": {
        "tag": "rag_pipeline_demo",
        "description": "Demo pipeline Using Azure OpenAI Connector for RAG application",
        "model_id": "EzpM0I0BVbLa0eH3b3an",
        "context_field_list": ["name","desc"],
        "system_prompt": "You are a helpful assistant",
        "user_instructions": "Generate a concise and informative answer in less than 100 words for the given question"
      }
    }
  ]
}
```

But the setting `plugins.ml_commons.rag_pipeline_feature_enabled` does not take effect when restarting the cluster and will throw an exception:

```
[2024-03-01T16:54:59,436][WARN ][o.o.s.p.SearchPipelineService] [opensearch-cluster_manager] failed to update search pipelines
org.opensearch.OpenSearchParseException: Error updating pipeline with id [rag_demo]
        at org.opensearch.search.pipeline.SearchPipelineService.innerUpdatePipelines(SearchPipelineService.java:198) ~[opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.search.pipeline.SearchPipelineService.applyClusterState(SearchPipelineService.java:156) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:608) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:596) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:563) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:486) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:188) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:854) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:283) [opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:246) [opensearch-2.12.0.jar:2.12.0]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) [?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) [?:?]
        at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: org.opensearch.ml.common.exception.MLException: plugins.ml_commons.rag_pipeline_feature_enabled is not enabled.
        at org.opensearch.searchpipelines.questionanswering.generative.GenerativeQAResponseProcessor$Factory.create(GenerativeQAResponseProcessor.java:324) ~[?:?]
        at org.opensearch.searchpipelines.questionanswering.generative.GenerativeQAResponseProcessor$Factory.create(GenerativeQAResponseProcessor.java:248) ~[?:?]
        at org.opensearch.search.pipeline.PipelineWithMetrics.readProcessors(PipelineWithMetrics.java:157) ~[opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.search.pipeline.PipelineWithMetrics.create(PipelineWithMetrics.java:98) ~[opensearch-2.12.0.jar:2.12.0]
        at org.opensearch.search.pipeline.SearchPipelineService.innerUpdatePipelines(SearchPipelineService.java:180) ~[opensearch-2.12.0.jar:2.12.0]
        ... 12 more
```

I tried to reset it to true, but it didn't work.

 Surprisingly, I first set it to false, then set it to true, and it can work again without throwing any exceptions.

```
PUT _cluster/settings
{
  "persistent": {
    "plugins.ml_commons.rag_pipeline_feature_enabled": false
  }
}
PUT _cluster/settings
{
  "persistent": {
    "plugins.ml_commons.rag_pipeline_feature_enabled": true
  }
}
```

**How can one reproduce the bug?**

1. Setting `plugins.ml_commons.rag_pipeline_feature_enabled` value to true
2. Restarting Opensearch cluster
3. View cluster startup logs

**What is the expected behavior?**

Setting Parameter `rag_pipeline_feature_enabled: true` is effect when restarting the cluster

**What is your host/environment?**
 - OS: Linux CentOS 7.9
 - Version: 2.12.0
 - Plugins: ml_commons

**Do you have any screenshots?**

**Do you have any additional context?**
Add any other context about the problem.

**Comment by dhrubo-os** [comment]
@HenryL27 could you please take a look at this issue?

**Comment by austintlee** [comment]
Fixed in 2.13.
